src/charm.py

```diff
@@ -111,6 +114,8 @@ def __init__(self, *args):
             self.on.tempo_pebble_custom_notice, self._on_tempo_pebble_custom_notice
         )
         self.framework.observe(self.on.update_status, self._on_update_status)
+        self.framework.observe(self.s3_requirer.on.credentials_changed, self._on_s3_changed)
+        self.framework.observe(self.s3_requirer.on.credentials_gone, self._on_s3_changed)
```

Comment:
I understand that this works for monolithic, but maybe we should start thinking of making S3 a requirement if tempo is replicated, please ignore this if this is already in plan for the tempo coordinator/worker

Reply:
that is indeed how it should be. let me check if it's already implemented...

Reply:
nope, doing it now.